### PR TITLE
Create CONTRIBUTING.md

### LINKED ISSUE #23: Create a roadmap for contributors
Create a `Contributing.md` file (which is standard for Open Source projects) that clearly delimits:

1. Project architecture and structure
2. Technologies used and their motives
3. Guidelines for development

### PR DESCRIPTION
**Pull request by kenan-rhoton**
Resolves #23 and clarifies the project architecture, structure and contributing guidelines (as well as acceptance criteria).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Project architecture
+
+The project is divided (as of now) into two main sections: *backend* and *frontend*, each with their quirks.
+
+## Backend architecture
+
+We use [Clojure](https://www.braveclojure.com) as the backend language for pretty much **everything** due to basically two reasons:
+
+1. Because of the "everything is immutable" paradigm, it's very easy to reason about the logic of the code, which will probably get very complex as time goes on.
+2. Due to its functional nature, it is *very* easy to run tests on it and a strict TDD discipline.
```
Comment (Masclins) [inline CONTRIBUTING.md]: I'd appreciated not having to google *TDD*.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+But some things also happen at the root:
+
+- **docker-compose.yml**: Defines how the different parts of the project launch and interact with each other.
+- **.travis.yml**: Defines how the different pars of the project are to be tested.
+
+However, we will understand this further in the following sections.
+
+## Backend structure
+
+- **backend/Dockerfile**: Defines how the Docker image for the backend will be built and run
```
Comment (Masclins) [inline CONTRIBUTING.md]: What's a *Docker image*?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+- **backend/project.clj**: Defines the project dependencies and entry point
+- **backend/test/card_game/**: Contains all the unit tests for the code. Remember that a unit test must be done *before* the actual code.
+- **backend/src/card_game/**: Contains the code that will run in production, the actual logic.
+- **backend/src/card_game/persistence.clj**: This is the only file to interact directly with the database. All database calls *must* pass through here.
+- **backend/src/card_game/api_handler.clj**: This is the only file to interact directly with frontend. Any actions to take as a response to a frontend call *must* be defined here.
+
+## Frontend structure
+
+- **frontend/Dockerfile** and **frontend/entry.sh**: Defines the Docker image and its entrypoint.
```
Comment (Masclins) [inline CONTRIBUTING.md]: Same as before: What's a *Docker image* and an *entrypoint* (maybe I should know, sorry for that)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+- **frontend/package.json** and **frontend/yarn.lock**: Define the project dependencies (`yarn.lock` is autogenerated by `yarn`)
+- **frontend/test/unit/**: Contains all the unit tests for the code. Remember that a unit test must be done *before* the actual code.
+- **frontend/public/**: The "website", with `index.html` being the entrypoint, but the real magic happens elsewhere...
+- **frontend/src/**: Contains the code that will run in production, with its own structure.
+- **frontend/src/main.js**: The official entrypoint for Vue. Should only be modified if we add a dependency that works through `Vue.use(dependency)`
+- **frontend/src/App.vue**: The main Vue component. Since we're mostly using everything through `router` this mainly hosts the `<router-view/>` and little else.
```
Comment (Masclins) [inline CONTRIBUTING.md]: I still have no clue of what **frontend/src/App.vue** does. I don't get what means we are using `router` nor what `<router-view\>` is.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+- **frontend/src/components/** and **frontend/src/views**: The application logic, separated into "views" and "components" following no good criteria as of yet, we will improve on this in the future.
+
+
+# Contributing guidelines
+
+For any change or contribution, this is what is expected of you:
+
+1. Announce within an issue that you want it assigned to you (or open said issue yourself).
+2. Do not take on more than one issue at a time
+3. Design the tests for your issue (unit tests always and in some cases e2e tests may be required)
```
Comment (Masclins) [inline CONTRIBUTING.md]: I would be very clear with the *one-test-at-a-time* philosophy

1. Make **a single** test that fails
2. Write **as few code as possible** to make it pass the test (and of course the already written tests)
3. Repeat from 1

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+We will value the following things on your contribution *in this order*:
+
+1. The tests pass
+2. The code has tests
+3. The tests are well-designed
+4. The code is readable
+5. The code is well-organized
+6. The code works and fixes the issue
+
+If at any point you have a question, please ask it if possible through a comment so others may benefit from the answer.
```
Comment (Masclins) [inline CONTRIBUTING.md]: All *Merges* are done through *Squash & merge*, so make sure to make the pull-request from a branch, not from your master.

**Comment by kenan-rhoton** [comment]
Do you mean doubts other than the ones you commented on? If so, it would be helpful to know them 🙃

**Pull request merged**